guardian/pinboard
# Add telemetry to pinboard

## What does this change?
Add telemetry to Pinboard to measure user activities. This PR makes use of the existing ed tools telemetry client ([github](https://github.com/guardian/editorial-tools-user-telemetry-service)) which sends the data to central ELK. We are building the Grafana [dashboard](https://metrics.gutools.co.uk/d/b-tyvbs7z/pinboard?orgId=1&refresh=5m&var-Stage=CODE&var-Interval=$__auto_interval_Interval) for visual representation of the collected data.


## How to test
1) Run this branch either locally or deploy on CODE. 
2) Open Composer/Grid CODE. You should see the Pinboard floaty on the lower right corner. 
3) When you interact with Pinboard* such as:
- opening / closing the Pinboard floaty
- sending messages
- adding grid images
- changing notification settings
...and other activities defined in [Telemetry.ts](https://github.com/guardian/pinboard/blob/4cd16d460eff8eb2a76deae53f4dfaa85d81f347/client/src/types/Telemetry.ts), you should see in the Network tab of the developer console that the activity data gets sent to `https://user-telemetry.code.dev-gutools.co.uk/event`.

*You can create a test article in Workflow CODE or use existing test articles' Pinboards to test sending messages etc.

** You may notice that multiple `'MESSAGE_SEEN'` events get sent for the same event. This is a known 'bug' that we aim to fix in the future but should not affect our Telemetry data as we're only capturing this event once per user in the Dashboard.

Example:
<img width="1539" alt="image" src="https://user-images.githubusercontent.com/47318984/172857863-3ee2d174-e391-431e-a779-2a321e8ddaea.png">
